processing/dwf_calculation_algorithm.py

```diff
@@ -0,0 +1,396 @@
+    ]
+
+    number_of_whole_hours = list(range(1, remaining_hours + 1))
+    # Loop over the remaining whole hours and append the new timesteps and their corresponding dwf factors
+    for h in number_of_whole_hours:
+        new_timestep = dwfFactorPerTimestep[-1][0] + 3600
+        newFactorHour = starting_hour + h + 1
+        new_factor = dwfFactors[newFactorHour % 24][1]
+        dwfFactorPerTimestep.append([new_timestep, new_factor])
+
+    # Append last timestep if there are seconds in the last hour
+    if secs_in_last_hour > 0:
+        dwfFactorPerTimestep.append(
+            [new_timestep + secs_in_last_hour, dwfFactors[(newFactorHour + 1) % 24][1]]
+        )
+
+    # Initialize list that will hold JSON
+    dwf_list = []
+    
+    # Generate JSON for each connection node
+    for i in enumerate(dwf_on_each_node):
+
+        # print(dwf_on_each_node[i[0]][1])
+        #dwfPerTimeStep = [
+        #    [row[0], row[1] * dwf_on_each_node[i[0]][1]] for row in dwfFactorPerTimestep
+        #]
+        
+        dwfPerTimeStep = """"""
+        for row in dwfFactorPerTimestep:
+            dwfPerTimeStep = dwfPerTimeStep + str(row[0]) + ',' + str(dwf_on_each_node[i[0]][1] * row[1]) + '\n'
+
+        dwfPerTimeStep = dwfPerTimeStep[:-1]
+        dwf_list.append(
+            {
+                "offset": 0,
+                "interpolate": 0,
+                "values": dwfPerTimeStep,
+                "units": "m3/s",
+                "connection_node": dwf_on_each_node[i[0]][0],
+            }
+        )
+
+    return dwf_list
+
+def dwf_json_to_csv(dwf_list, output_csv_file):
+        
+    # INPUT CSV FOR API CLIENT :
+    # FORMAT : lat_id, connection_node_id, timeseries
+    with open(output_csv_file,'w', newline='') as csv_file: 
+        writer = csv.writer(csv_file)
+        for i, row in enumerate(dwf_list):
+            lat_id = i
+            connection_node_id = row['connection_node']
+            timeseries = row['values']
+            writer.writerow([str(lat_id), str(connection_node_id), timeseries])
+
+def get_sec(time_str):
+    """Get Seconds from time."""
+    m, s = time_str.split(":")
+    return int(m) * 60 + int(s)
+
+
+class DWFCalculatorAlgorithm(QgsProcessingAlgorithm):
+    """
+    This is an example algorithm that takes a vector layer and
+    creates a new identical one.
+
+    It is meant to be used as an example of how to create your own
+    algorithms and explain methods and variables used to do it. An
+    algorithm like this will be available in all elements, and there
+    is not need for additional work.
+
+    All Processing algorithms should extend the QgsProcessingAlgorithm
+    class.
+    """
+
+    # Constants used to refer to parameters and outputs. They will be
+    # used when calling the algorithm from another algorithm, or when
+    # calling from the QGIS console.
+
+    OUTPUT = 'OUTPUT'
+    INPUT = 'INPUT'
+
+    def initAlgorithm(self, config):
```

Comment:
In places like this, you cannot escape qgis's mixedCase naming style, so that's fine.
But in the rest of the code: underscores.